website/project/views/comment.py

```diff
@@ -190,6 +294,53 @@ def add_comment(**kwargs):
     }, http.CREATED
 
 
+def get_file_provider(page, root_target):
+    if page == 'files':
+        if root_target.provider == 'googledrive':
+            return 'Google Drive'
+        elif root_target.provider == 'osfstorage':
+            return 'OSF storage'
+        else:
+            return root_target.provider.title()
+    else:
+        return ''
+
+
+def get_page_type(page, node):
+    if page == 'wiki':
+        return 'wiki'
+    elif page == 'files':
+        return 'file'
+    elif node.parent_node:
+        return 'component'
+    else:
+        return 'project'
+
+
+def get_root_target_title(page, root_target):
+    if page == 'wiki':
+        return root_target.page_name
+    elif page == 'files':
+        metadata = requests.get(root_target.metadata_url).json()
```

Comment:
As discussed briefly in-person, it may be preferable to use `FileGuid#enrich()` method here, which will automatically cache results and throw a proper error.